From 4e13126996de5f53e576a77a677f1ffa2b726599 Mon Sep 17 00:00:00 2001
From: SotrForgotten <i@omc.hk>
Date: Wed, 27 Jul 2016 23:34:47 +0800
Subject: [PATCH] SPIGOT-1541: Fix firework launching


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftFirework.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftFirework.java
index 76a9bdb..771ff68 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftFirework.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftFirework.java
@@ -3,6 +3,7 @@ package org.bukkit.craftbukkit.entity;
 import net.minecraft.server.EntityFireworks;
 import net.minecraft.server.ItemStack;
 import net.minecraft.server.Items;
+import com.google.common.base.Optional;
 
 import org.bukkit.Material;
 import org.bukkit.craftbukkit.CraftServer;
@@ -14,7 +15,6 @@ import org.bukkit.inventory.meta.FireworkMeta;
 import java.util.Random;
 
 public class CraftFirework extends CraftEntity implements Firework {
-    private static final int FIREWORK_ITEM_INDEX = 8;
 
     private final Random random = new Random();
     private final CraftItemStack item;
@@ -26,7 +26,7 @@ public class CraftFirework extends CraftEntity implements Firework {
 
         if (item == null) {
             item = new ItemStack(Items.FIREWORKS);
-            getHandle().getDataWatcher().watch(FIREWORK_ITEM_INDEX, item);
+            getHandle().getDataWatcher().set(EntityFireworks.FIREWORK_ITEM, Optional.of(item));
         }
 
         this.item = CraftItemStack.asCraftMirror(item);
-- 
2.8.2.windows.1

