From 031252103a6bde720c7bb6229d7ec9bd1621a644 Mon Sep 17 00:00:00 2001
From: SotrForgotten <i@omc.hk>
Date: Thu, 28 Jul 2016 00:06:41 +0800
Subject: [PATCH] SPIGOT-1573: Fix command blocks after respawning


diff --git a/src/main/java/net/minecraft/server/PlayerList.java b/src/main/java/net/minecraft/server/PlayerList.java
index 3c2d319..beee5ac 100644
--- a/src/main/java/net/minecraft/server/PlayerList.java
+++ b/src/main/java/net/minecraft/server/PlayerList.java
@@ -566,7 +566,7 @@ public abstract class PlayerList {
                 blockposition1 = EntityHuman.getBed(cworld.getHandle(), blockposition, flag1);
                 if (blockposition1 != null) {
                     isBedSpawn = true;
-                    location = new Location(cworld, blockposition1.getX() + 0.5, blockposition1.getY(), blockposition1.getZ() + 0.5);
+                     location = new Location(cworld, (double) ((float) blockposition1.getX() + 0.5F), (double) ((float) blockposition1.getY() + 0.1F), (double) ((float) blockposition1.getZ() + 0.5F));
                 } else {
                     entityplayer1.setRespawnPosition(null, true);
                     entityplayer1.playerConnection.sendPacket(new PacketPlayOutGameStateChange(0, 0.0F));
@@ -576,7 +576,7 @@ public abstract class PlayerList {
             if (location == null) {
                 cworld = (CraftWorld) this.server.server.getWorlds().get(0);
                 blockposition = cworld.getHandle().getSpawn();
-                location = new Location(cworld, blockposition.getX() + 0.5, blockposition.getY(), blockposition.getZ() + 0.5);
+				location = new Location(cworld, (double) ((float) blockposition.getX() + 0.5F), (double) ((float) blockposition.getY() + 0.1F), (double) ((float) blockposition.getZ() + 0.5F));
             }
 
             Player respawnPlayer = cserver.getPlayer(entityplayer1);
@@ -618,6 +618,7 @@ public abstract class PlayerList {
         entityplayer1.playerConnection.sendPacket(new PacketPlayOutSpawnPosition(blockposition1));
         entityplayer1.playerConnection.sendPacket(new PacketPlayOutExperience(entityplayer1.exp, entityplayer1.expTotal, entityplayer1.expLevel));
         this.b(entityplayer1, worldserver);
+		this.f(entityplayer1);
 
         if (!entityplayer.playerConnection.isDisconnected()) {
             worldserver.getPlayerChunkMap().addPlayer(entityplayer1);
@@ -626,6 +627,8 @@ public abstract class PlayerList {
             this.playersByName.put(entityplayer1.getName(), entityplayer1); // Spigot
             this.j.put(entityplayer1.getUniqueID(), entityplayer1);
         }
+		// entityplayer1.syncInventory();
+        entityplayer1.setHealth(entityplayer1.getHealth());
         // Added from changeDimension
         updateClient(entityplayer); // Update health, etc...
         entityplayer.updateAbilities();
@@ -633,11 +636,7 @@ public abstract class PlayerList {
             MobEffect mobEffect = (MobEffect) o1;
             entityplayer.playerConnection.sendPacket(new PacketPlayOutEntityEffect(entityplayer.getId(), mobEffect));
         }
-        // entityplayer1.syncInventory();
-        // CraftBukkit end
-        entityplayer1.setHealth(entityplayer1.getHealth());
 
-        // CraftBukkit start
         // Don't fire on respawn
         if (fromWorld != location.getWorld()) {
             PlayerChangedWorldEvent event = new PlayerChangedWorldEvent(entityplayer.getBukkitEntity(), fromWorld);
-- 
2.8.2.windows.1

