From d26954aa26833e38c95c4f59084b66b294889ce2 Mon Sep 17 00:00:00 2001
From: SotrForgotten <i@omc.hk>
Date: Thu, 28 Jul 2016 00:10:36 +0800
Subject: [PATCH] SPIGOT-1823: Use fast chunk lookup in CraftWorld also


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
index 958fe4f..dbda3c9 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
@@ -141,7 +141,7 @@ public class CraftWorld implements World {
     // PaperSpigot end
 
     public Chunk getChunkAt(int x, int z) {
-        return this.world.chunkProviderServer.getChunkAt(x, z).bukkitChunk;
+		return this.world.chunkProviderServer.getOrCreateChunkFast(x, z).bukkitChunk;
     }
 
     public Chunk getChunkAt(Block block) {
-- 
2.8.2.windows.1

