From 0c14dc8524c32a57b2628b66b6bb175608f2a5bf Mon Sep 17 00:00:00 2001
From: SotrForgotten <i@omc.hk>
Date: Thu, 28 Jul 2016 00:14:02 +0800
Subject: [PATCH] SPIGOT-1875: Broken leashes not dropping


diff --git a/src/main/java/net/minecraft/server/EntityInsentient.java b/src/main/java/net/minecraft/server/EntityInsentient.java
index b82b984..20c8524 100644
--- a/src/main/java/net/minecraft/server/EntityInsentient.java
+++ b/src/main/java/net/minecraft/server/EntityInsentient.java
@@ -846,7 +846,9 @@ public abstract class EntityInsentient extends EntityLiving {
             this.bo = false;
             this.bp = null;
             if (!this.world.isClientSide && flag1) {
+				this.forceDrops = true; // CraftBukkit
                 this.a(Items.LEAD, 1);
+				this.forceDrops = false; // CraftBukkit
             }
 
             if (!this.world.isClientSide && flag && this.world instanceof WorldServer) {
-- 
2.8.2.windows.1

