From 3ad481f60d6c7d53e33dcaa88ab3e00136bdd2d1 Mon Sep 17 00:00:00 2001
From: SotrForgotten <i@omc.hk>
Date: Fri, 5 Aug 2016 23:30:24 +0800
Subject: [PATCH] Fix entity location


diff --git a/src/main/java/net/minecraft/server/Chunk.java b/src/main/java/net/minecraft/server/Chunk.java
index 590fab6..64ba9f7 100644
--- a/src/main/java/net/minecraft/server/Chunk.java
+++ b/src/main/java/net/minecraft/server/Chunk.java
@@ -718,11 +718,12 @@ public class Chunk {
 
         if (i != this.locX || j != this.locZ) {
             // CraftBukkit start
-            Bukkit.getLogger().warning("Wrong location for " + entity + " in world '" + world.getWorld().getName() + "'!");
+            // Bukkit.getLogger().warning("Wrong location for " + entity + " in world '" + world.getWorld().getName() + "'!");
             // Chunk.c.warn("Wrong location! (" + i + ", " + j + ") should be (" + this.locX + ", " + this.locZ + "), " + entity, new Object[] { entity});
-            Bukkit.getLogger().warning("Entity is at " + entity.locX + "," + entity.locZ + " (chunk " + i + "," + j + ") but was stored in chunk " + this.locX + "," + this.locZ);
+            // Bukkit.getLogger().warning("Entity is at " + entity.locX + "," + entity.locZ + " (chunk " + i + "," + j + ") but was stored in chunk " + this.locX + "," + this.locZ);
             // CraftBukkit end
-            entity.die();
+            // entity.die();
+			entity.setPosition(this.locX, entity.locY, this.locZ); // Torch
         }
 
         int k = MathHelper.floor(entity.locY / 16.0D);
-- 
2.8.2.windows.1

