From 45e77c82f3c49449e4314df0efc04785a18f854c Mon Sep 17 00:00:00 2001
From: SotrForgotten <i@omc.hk>
Date: Tue, 16 Aug 2016 13:28:50 +0800
Subject: [PATCH] Fork packet send


diff --git a/pom.xml b/pom.xml
index a468d4e..cbbf880 100644
--- a/pom.xml
+++ b/pom.xml
@@ -265,15 +265,6 @@
                     </excludes>
                 </configuration>
             </plugin>
-			  <!-- Torch - Do not deploy the server -->
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-deploy-plugin</artifactId>
-                <version>2.8.2</version>
-                <configuration>
-                    <skip>true</skip>
-                </configuration>
-            </plugin>
         </plugins>
     </build>
 </project>
diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index 96b04f9..85dabca 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -23,6 +23,7 @@ import org.apache.logging.log4j.Logger;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
 import java.util.HashSet;
+import java.util.concurrent.RecursiveAction;
 
 import org.bukkit.craftbukkit.entity.CraftPlayer;
 import org.bukkit.craftbukkit.event.CraftEventFactory;
@@ -889,8 +890,35 @@ public class PlayerConnection implements PacketListenerPlayIn, IUpdatePlayerList
         }
 
     }
-
-    public void sendPacket(final Packet packet) {
+	
+	// Torch start
+	class SendTask extends RecursiveAction {
+		Packet<?> packet;
+
+		SendTask(Packet<?> packet) {
+			this.packet = packet;
+		}
+
+		@Override
+		protected void compute() {
+			sendPacket_sub(packet);
+		}
+	}
+
+	public void sendPacket(final Packet packet) {
+		if (packet == null || this.processedDisconnect) { // Spigot
+			return;
+		}
+		if (packet instanceof PacketPlayOutRespawn) {
+			this.networkManager.handle(packet);
+			return;
+		}
+		
+		SendTask task = new SendTask(packet);
+		task.fork();
+	}
+
+    public void sendPacket_sub(final Packet packet) {
         if (packet instanceof PacketPlayOutChat) {
             PacketPlayOutChat packetplayoutchat = (PacketPlayOutChat) packet;
             EntityHuman.EnumChatVisibility entityhuman_enumchatvisibility = this.player.getChatFlags();
@@ -905,9 +933,8 @@ public class PlayerConnection implements PacketListenerPlayIn, IUpdatePlayerList
         }
 
         // CraftBukkit start
-        if (packet == null || this.processedDisconnect) { // Spigot
-            return;
-        } else if (packet instanceof PacketPlayOutSpawnPosition) {
+		// Torch end
+        if (packet instanceof PacketPlayOutSpawnPosition) {
             PacketPlayOutSpawnPosition packet6 = (PacketPlayOutSpawnPosition) packet;
             this.player.compassTarget = new Location(this.getPlayer().getWorld(), packet6.position.getX(), packet6.position.getY(), packet6.position.getZ());
         }
-- 
2.8.2.windows.1

